builder error using anonymous sumtype with none

**V version:** V 0.3.0 b01f71d.3656d44
**OS:** 64-bit linux

**What did you do?**

```v
fn number() int | none {
	return none
}

n := number()
match n {
	int { println('n: $n') }
	none { println('?') }
}
```

and

```v
fn number() int | none {
	return none
}

n := number()
if n is none {
	println('?')
} else {
	println('n: $n')
}
```

**What did you expect to see?**

```
?
```
 
**What did you see instead?**

```
/usr/lib64/crt1.o: error: Invalid relocation entry [15] '.rela.debug_info' @ 000004f5
tcc: error: file 'crt1.o' not found
/tmp/v_1000/bar.4267398468021436734.tmp.c:12224: error: invalid operand types for binary operation
builder error: 
==================
C error. This should never happen.
```

And when compiling the .c file:

```
foo.c:12206:25: error: invalid operands to binary == (have ‘int’ and ‘IError’)
12206 |         else if (n._typ == _const_none__) {
      |                  ~~~~~~ ^~
      |                   |
      |                   int
```

for the first example and this for the second:

```
bar.c:12222:22: error: invalid operands to binary == (have ‘int’ and ‘IError’)
12222 |         if ((n)._typ == _const_none__) {
      |             ~~~~~~~~ ^~
      |                |
      |                int
```

--- a/vlib/v/gen/c/match.v
+++ b/vlib/v/gen/c/match.v
@@ -172,7 +172,11 @@ fn (mut g Gen) match_expr_sumtype(node ast.MatchExpr, is_expr bool, cond_var str
 				be := unsafe { &branch.exprs[sumtype_index] }
 				if sym.kind == .sum_type {
 					g.write('${dot_or_ptr}_typ == ')
-					g.expr(be)
+					if be is ast.None {
+						g.write('$ast.none_type.idx() /* none */')
+					} else {
+						g.expr(be)
+					}
 				} else if sym.kind == .interface_ {
 					if be is ast.TypeNode {
 						typ := be as ast.TypeNode

--- a/vlib/v/tests/match_sumtype_var_with_none_test.v
+++ b/vlib/v/tests/match_sumtype_var_with_none_test.v
@@ -1,0 +1,13 @@
+fn number() int|none {
+	return none
+}
+
+fn test_match_sumtype_var_with_none() {
+	n := number()
+	ret := match n {
+		int { 'n: $n' }
+		none { '?' }
+	}
+	println(ret)
+	assert ret == '?'
+}

